framework/src/utils/GriddedData.C

```diff
@@ -57,6 +57,65 @@ GriddedData::GriddedData(std::string file_name)
   parse(_dim, _axes, _grid, _fcn, _step, file_name);
 }
 
+GriddedData::GriddedData(std::vector<Real> x_axis,
+                         std::vector<Real> y_axis,
+                         std::vector<Real> z_axis,
+                         std::vector<Real> t_axis,
+                         std::vector<Real> fcn)
+{
+  unsigned int num_data_pt = 1;
+  int ind = 0;
+  _axes.resize(0);
+  _fcn = fcn;
+  _dim = 0;
+  if (x_axis.size() != 0)
+  {
+    ind = 0;
+    updateGrid(num_data_pt, x_axis, ind);
+  }
+  if (y_axis.size() != 0)
+  {
+    ind = 1;
+    updateGrid(num_data_pt, y_axis, ind);
+  }
+  if (z_axis.size() != 0)
+  {
+    ind = 2;
+    updateGrid(num_data_pt, z_axis, ind);
+  }
+  if (t_axis.size() != 0)
+  {
+    ind = 3;
+    updateGrid(num_data_pt, t_axis, ind);
+  }
+  if (_dim == 0)
+    mooseError("No valid AXIS lines found by GriddedData");
+  if (num_data_pt != _fcn.size())
+    mooseError("According to the provided axes and function values, number of data points is ",
+               num_data_pt,
+               " but ",
+               _fcn.size(),
+               " function values were given by user");
+  // step is useful in evaluateFcn
+  _step.resize(_dim);
+  _step[0] = 1; // this is actually not used
+  for (unsigned int i = 1; i < _dim; ++i)
```

Comment:
can you compute num_data_pt in this loop and not compute it in the update grid function.